os/fs/smartfs/smartfs_utils.c

```diff
@@ -739,20 +739,25 @@ int smartfs_sync_internal(struct smartfs_mountpt_s *fs, struct smartfs_ofile_s *
 		if (sf->entry.datalen > 0 && used_value == 0) {
 			used_value = fs->fs_llformat.availbytes - sizeof(struct smartfs_chain_header_s);
 		}
-
 		set_used_byte_count((uint8_t *)header->used, used_value);
+
+		smartfs_setbuffer(&readwrite, sf->currsector, offsetof(struct smartfs_chain_header_s, used), sizeof(uint16_t), (uint8_t *)&fs->fs_rwbuffer[offsetof(struct smartfs_chain_header_s, used)]);
```

Comment:
Here also we should check something more...
It is too complicated code now,  actually writing entire of chain header doesn't takes such a long time than write used only, Let's change to write entire of chain header.....simply...